src/transactions/TransactionMetaFrame.cpp

```diff
@@ -0,0 +1,284 @@
+// Copyright 2022 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/TransactionMetaFrame.h"
+#include "crypto/SHA.h"
+#include "transactions/TransactionFrameBase.h"
+#include "util/GlobalChecks.h"
+#include "util/ProtocolVersion.h"
+#include <xdrpp/xdrpp/marshal.h>
+
+namespace stellar
+{
+TransactionMetaFrame::TransactionMetaFrame(uint32_t protocolVersion)
+{
+    // The TransactionMeta v() switch can be in 4 positions 0, 1, 2, 3. We
+    // do not support 0 or 1 at all -- core does not produce it anymore and we
+    // have no obligation to consume it under any circumstance -- so this
+    // class just switches between cases 2 and 3.
+    mVersion = 2;
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    if (protocolVersionStartsFrom(protocolVersion, SOROBAN_PROTOCOL_VERSION))
+    {
```

Comment:
you forgot to update `normalizeMeta` in dumpxdr.cpp

Reply:
mmm actually for hashes to work properly we have to normalize the meta in protocol (so here)

Reply:
(and I'm not sure this can be done more efficiently that what was done in dumpxdr)

Reply:
fixed